src/app/sessions/containers/sessions-listings-search/sessions-listings-search.component.ts

```diff
@@ -98,45 +100,56 @@ export class SessionsListingsSearchComponent implements OnInit {
             .filter(s => this.sessionsFilterService.filterByUtilization(s, filters.utilization));
     }
 
-    selectHearingPart(hearingPart: HearingViewmodel) {
-        this.selectedHearingPart = hearingPart;
+    selectHearing(hearing: HearingViewmodel) {
+        this.selectedHearing = hearing;
+        this.numberOfSessionsNeeded = hearing !== undefined ? hearing.numberOfSessionsNeeded : 0;
     }
 
-    assignToSession(assignHearingData: AssignHearingData) {
-        this.hearingModificationService.assignWithSession({
-            hearingId: this.selectedHearingPart.id,
-            hearingVersion: this.selectedHearingPart.version,
-            sessionId: this.selectedSession.id,
-            sessionVersion: this.selectedSession.version,
+    selectSession(sessions: SessionViewModel[]) {
+        this.selectedSessions = sessions;
+    }
+
+    assignToSessions(assignHearingData: AssignHearingData) {
+        let assignment = {
+            hearingId: this.selectedHearing.id,
+            hearingVersion: this.selectedHearing.version,
+            sessionsData: this.selectedSessions.map(session => {return {sessionId: session.id, sessionVersion: session.version}}),
             userTransactionId: uuid(),
-            start: moment(assignHearingData.startTime, 'HH:mm').toDate()
-        } as HearingToSessionAssignment);
+            start: this.selectedSessions.length > 1 ? null : moment(assignHearingData.startTime, 'HH:mm').toDate()
+        } as HearingToSessionAssignment;
+
+        this.hearingModificationService.assignWithSession(assignment);
 
         this.openSummaryDialog().afterClosed().subscribe((accepted) => {
             this.store.dispatch(new fromHearingPartsActions.Search());
             if (accepted) {
                 this.store.dispatch(new fromNotes.CreateMany(assignHearingData.notes));
             }
 
-            this.selectedHearingPart = undefined;
-            this.selectedSession = undefined;
+            this.selectedHearing = undefined;
+            this.selectedSessions = undefined;
         });
     }
 
-    selectSession(session: SessionViewModel) {
-        this.selectedSession = session;
-    }
-
     assignButtonEnabled() {
-        return !!(safe(() => this.selectedHearingPart.id) && (safe(() => this.selectedSession.id)));
+        if (this.selectedHearing === undefined) {
+            this.errorMessage = '';
+            return false;
+        } else if (this.numberOfSessionsNeeded === this.selectedSessions.length) {
+            this.errorMessage = '';
+            return this.checkIfOnlyOneJudgeSelected();
+        } else {
+            this.errorMessage = 'Incorrect number of sessions selected';
+            return false;
+        }
     }
 
     openAssignDialog() {
         this.dialog.open(AssignHearingDialogComponent, {
```

Comment:
if you will create interface for data property, you can utilize here as well eg:
```suggestion
        this.dialog.open<any, YourSuperInterface>(AssignHearingDialogComponent, {
```